pkg/backend/token_authcode.go

```diff
@@ -41,29 +38,41 @@ type refreshDescriptor struct {
 var _ scheduler.Descriptor = &refreshDescriptor{}
 
 func (rd *refreshDescriptor) Run(ctx context.Context, pc chan<- scheduler.Process) error {
-	ticker := rd.backend.clock.NewTicker(refreshInterval)
-	defer ticker.Stop()
+	// start out checking once per minute
+	refreshInterval := 60
+	ticker := rd.backend.clock.NewTicker(time.Duration(refreshInterval) * time.Second)
 
 	for {
-		err := rd.backend.data.Managers(rd.storage).AuthCode().ForEachAuthCodeKey(ctx, func(keyer persistence.AuthCodeKeyer) {
-			proc := &refreshProcess{
-				backend: rd.backend,
-				storage: rd.storage,
-				keyer:   keyer,
+		c, err := rd.backend.getCache(ctx, rd.storage)
+		if err == nil && c != nil && c.Config.RefreshInterval != 0 {
```

Comment:
One case that I'm thinking about is what happens if someone disables the refreshing and then wants to subsequently re-enable it. I think, just reading this, that the check to re-enable will occur at whatever the last-used refresh interval was. So consider someone setting it to 86400, then 0, then 60. It won't run again until a day has passed (or the plugin restarts, which is also confusing?).

It would be nice to be able to somehow "wake up" this descriptor again whenever the config changes so we can re-evaluate what we should do. The `invalidate()` method in `lifecycle.go` takes care of that somewhat right now. I'll do some noodling on what options we have here and get back to you too.

Reply:
That issue occurred to me but I thought it would be so rare that it wasn't worth worrying about or doing anything that would affect performance such as always checking at most once every minute.